iac-extensions/spring-config/src/main/java/org/sonar/iac/springconfig/checks/DebugFeatureEnabledCheck.java

```diff
@@ -0,0 +1,44 @@
+/*
+ * SonarQube IaC Plugin
+ * Copyright (C) 2021-2024 SonarSource SA
+ * mailto:info AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+package org.sonar.iac.springconfig.checks;
+
+import java.util.Locale;
+import java.util.Set;
+import org.sonar.check.Rule;
+import org.sonar.iac.common.api.checks.CheckContext;
+import org.sonar.iac.springconfig.tree.api.Tuple;
+
+@Rule(key = "S4507")
+public class DebugFeatureEnabledCheck extends AbstractSensitiveKeyCheck {
+  private static final String MESSAGE = "Make sure this debug feature is deactivated before delivering the code in production.";
+  private static final Set<String> SENSITIVE_KEYS = Set.of("debug");
+
+  @Override
+  protected Set<String> sensitiveKeys() {
+    return SENSITIVE_KEYS;
+  }
+
+  @Override
+  protected void checkValue(CheckContext ctx, Tuple tuple, String value) {
+    if ("true".equals(value.toLowerCase(Locale.ROOT))) {
```

Comment:
```suggestion
    if ("true".equalsIgnoreCase(value)) {
```
Possible alternative.